File: Reqnroll/Chapter17/17.5-TemplatedPersonaPattern/After/WIMP.Specs/Support/CustomParameterTypes.cs
using Reqnroll;

namespace WIMP.Specs.Support;

[Binding]
public class CustomParameterTypes
{
    [StepArgumentTransformation(@"(\d+-\d+-\d+)")]
    public DateOnly ConvertDateOnly(string value)
    {
        return DateOnly.Parse(value);
    }
}

File: CucumberJVM/Chapter17/17.5-TemplatedPersonaPattern/After/src/test/java/com/wimp/app/specs/support/CustomParameterTypes.java
package com.wimp.app.specs.support;

import io.cucumber.java.DataTableType;
import io.cucumber.java.ParameterType;

import java.time.LocalDate;
import java.util.Map;

public class CustomParameterTypes {

    @DataTableType
    public SalesFilterData salesFilterDataRow(Map<String, String> row) {
        return new SalesFilterData(
            row.get("exclude section"));
    }

    @ParameterType("\\d{4}-\\d{2}-\\d{2}")
    public LocalDate date(String value) {
        return LocalDate.parse(value);
    }
}
